Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -13,84 +13,144 @@ References:
+                , ActorUserId = tostring(EventData.SubjectUserSid)
+                , ActorSessionId = tostring(EventData.SubjectLogonId)
+                , ActingProcessName = tostring(EventData.ProcessName)
+                , ActingProcessId = tostring(toint(tolong(EventData.ProcessId)))
+                , RegistryKey = iif(EventData.ObjectName startswith @"\REGISTRY\MACHINE", replace_string(tostring(EventData.ObjectName), @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                    , replace_string(tostring(EventData.ObjectName), @"\REGISTRY\USER","HKEY_USERS")
+                )
+        };
+        let ASIM_ParseSecurityEvents = (SecurityEvent:(SubjectDomainName:string,SubjectUserName:string,ProcessId:string,ObjectName:string,SubjectUserSid:string,SubjectLogonId:string,ProcessName:string)) {
+            SecurityEvent
+            | project-rename ActorUsername = SubjectUserName
+                , ActorUserId = SubjectUserSid
+                , ActorSessionId = SubjectLogonId
+                , ActingProcessName = ProcessName
+                , ActorDomainName = SubjectDomainName
+            | extend ActorUsername = iif(isnotempty(ActorDomainName), strcat(ActorDomainName, @'\', ActorUsername), ActorUsername)
+                , ActingProcessId = tostring(toint(tolong(ProcessId)))
+                , RegistryKey = iif(ObjectName startswith @"\REGISTRY\MACHINE", replace_string(ObjectName, @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                    , replace_string(ObjectName, @"\REGISTRY\USER","HKEY_USERS")
+                )
+        };
+        let Event4663TypeLookup = datatable (AccessMask:string,EventType:string)
+        [
+            "0x1", "RegistryValueRead"
+            , "0x10", "MetadataModified"
+            , "0x10000", "ObjectDeleted"
+            , "0x2", "RegistryValueSet"
+            , "0x20000", "MetadataAccessed"
+            , "0x20006", "MetadataAccessed"
+            , "0x40000", "MetadataModified"
+            , "0x8", "RegistrySubkeyEnumerated"
+            , "0x80000", "MetadataModified"
+        ];
+        let Event4567TypeLookup = datatable (EventOriginalSubType:string,EventType:string)
+        [
+            "%%1904", "RegistryValueSet"
+            , "%%1905", "RegistryValueSet"
+            , "%%1906", "RegistryValueDeleted"
+        ];
+        let RegistryType = datatable (TypeCode:string,TypeName:string)
+        [
+            "%%1872", "REG_NONE"
+            , "%%1873",	"REG_SZ"
+            , "%%1874", "REG_EXPAND_SZ"
+            , "%%1875", "REG_BINARY"
+            , "%%1876", "REG_DWORD"
+            , "%%1879", "REG_MULTI_SZ"
+            , "%%1883", "REG_QWORD"
+        ];
+        union isfuzzy=false (
+            WindowsEvent
+            | where EventID == 4663
+                and EventData.ObjectType == "Key"
+            | extend AccessMask = tostring(EventData.AccessMask)
+                , Type = "WindowsEvent"
+            | lookup Event4663TypeLookup on AccessMask
```

Comment:
Given the combinations we have seen, such as "20006", We may want to handle values not found in the list and assign the EventType "Other".